spec/backup.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { ResearchStudy } from '../src/fhir-types';
+import data from './data/resource.json'; //trial missing summary, inclusion/exclusion criteria, phase and study type
+import * as trialbackup from '../src/trialbackup';
+import * as fs from 'fs';
+
+describe('.findNCTNumber', () => {
+  it('finds an NCT number with the proper coding system', () => {
+    expect(trialbackup.findNCTNumber({
+      resourceType: 'ResearchStudy',
+      identifier: [
+        {
+          'system': 'other',
+          'value': 'ignoreme'
+        },
+        {
+          'system': trialbackup.CLINICAL_TRIAL_IDENTIFIER_CODING_SYSTEM_URL,
+          'value': 'test'
+        }
+      ]
+    })).toEqual('test');
+  });
+  it('finds an NCT number based on regexp', () => {
+    expect(trialbackup.findNCTNumber({
+      resourceType: 'ResearchStudy',
+      identifier: [
+        {
+          'system': 'other',
+          'value': 'ignoreme'
+        },
+        {
+          'system': 'invalidsystem',
+          'value': 'NCT12345678'
+        }
+      ]
+    })).toEqual('NCT12345678');
+  });
+  it('returns null with no valid NCT number', () => {
+    expect(trialbackup.findNCTNumber({
+      resourceType: 'ResearchStudy',
+      identifier: [
+        {
+          'system': 'other',
+          'value': 'ignoreme'
+        },
+        {
+          'system': 'invalid',
+          'value': 'alsoignored'
+        }
+      ]
+    })).toBeNull();
+  });
+  it('returns null with no identifier', () => {
+    expect(trialbackup.findNCTNumber({
+      resourceType: 'ResearchStudy',
+      identifier: [ ]
+    })).toBeNull();
+    expect(trialbackup.findNCTNumber({ resourceType: 'ResearchStudy' })).toBeNull();
+  });
+});
+
+describe('backup tests', () => {
+  let study: ResearchStudy = data.entry[0].resource as ResearchStudy;
+  const nctID = trialbackup.findNCTNumber(study);
+  if (nctID === null) {
+    throw new Error('ResearchStudy has no NCT number');
+  }
+  const nctIds = [ nctID ];
+  beforeEach(function () {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
+  });
+  afterEach(function () {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 5000;
+  });
+  const filepath = './src';
+  let checker : trialbackup.BackupSystem;
+  let trial : trialbackup.TrialBackup;
+  beforeAll(async function () {
+    
+    const downloader = new trialbackup.ClinicalTrialGov(filepath)
```

Comment:
We should figure out how to mock the download so that we can download a known-good version and aren't relying on a network connection and ClinicalTrial.gov being up.